Libraries/builtin/Sources/copy/Driver.cpp

```diff
@@ -41,42 +42,45 @@ name()
 static bool
 CopyPath(Filesystem *filesystem, std::string const &inputPath, std::string const &outputPath)
 {
-    if (!filesystem->createDirectory(FSUtil::GetDirectoryName(outputPath), true)) {
-        return false;
-    }
+    /* Must be writable once copied to allow subsequent builds to overwrite. */
+    Permissions permissions;
+    permissions.user(Permissions::Permission::Write, true);
+    Permissions::Operation operation = Permissions::Operation::Add;
+
+    if (filesystem->isDirectory(inputPath)) {
+        if (!filesystem->copyDirectory(inputPath, outputPath, true)) {
```

Comment:
I like multiple lines because it makes it easier if some lines return more, like `ext::optional<std::string>` or something.

Ideally, all of these functions would return `Result<void>` or `Result<std::string>` as appropriate, then use something either like Rust's `try!` to skip the verbose error handling, but I don't think that's possible in C++. :(